libraries/ESP8266WiFi/src/ESP8266WiFiSTA-WPS.cpp

```diff
@@ -0,0 +1,84 @@
+
+#include "ESP8266WiFi.h"
+#include "ESP8266WiFiGeneric.h"
+#include "ESP8266WiFiSTA.h"
+
+void wifi_wps_status_cb(wps_cb_status status);
+
+/**
+ * WPS config
+ * so far only WPS_TYPE_PBC is supported (SDK 1.2.0)
+ * @return ok
+ */
+bool beginWPSConfig(void) {

```

Comment:
You can keep this a member function of `ESP8266WiFiSTAClass`, jump move implementation to a separate file, as you did.